src/pages/ConvertVrt/Convert/index.tsx

```diff
@@ -1,50 +1,134 @@
 /** @jsxImportSource @emotion/react */
-import React from 'react';
+import React, { useCallback, useMemo } from 'react';
 import BigNumber from 'bignumber.js';
 import noop from 'noop-ts';
 import { Typography } from '@mui/material';
 import {
   FormikTokenTextField,
-  LabeledProgressBar,
   FormikSubmitButton,
   ConnectWallet,
+  EnableToken,
   Icon,
   TokenTextField,
 } from 'components';
+import useSuccessfulTransactionModal from 'hooks/useSuccessfulTransactionModal';
+import toast from 'components/Basic/Toast';
 import { useTranslation } from 'translation';
 import useConvertToReadableCoinString from 'hooks/useConvertToReadableCoinString';
-import { format } from 'utilities/common';
-import { formatI18nextRelativetimeValues } from 'utilities';
 import { AmountForm, ErrorCode } from 'containers/AmountForm';
+import { getContractAddress, formatI18nextRelativetimeValues } from 'utilities';
+import { InternalError } from 'utilities/errors';
+import { convertCoinsToWei, convertWeiToCoins, formatCoinsToReadableValue } from 'utilities/common';
 import { VRT_ID, XVS_ID, VRT_DECIMAL } from '../constants';
 import { useStyles } from '../styles';
 
-interface ConvertProps {
-  xvsTotalWei: BigNumber;
-  xvsToVrtRate: BigNumber;
-  vrtLimitUsedWei: BigNumber;
-  vrtLimitWei: BigNumber;
-  vrtConversionEndTime: string | undefined;
+export interface IConvertProps {
+  xvsToVrtConversionRatio: BigNumber | undefined;
+  vrtConversionEndTime: Date | undefined;
+  userVrtBalanceWei: BigNumber | undefined;
+  convertVrtLoading: boolean;
+  userVrtEnabled: boolean;
+  convertVrt: (amount: string) => Promise<string>;
   walletConnected: boolean;
 }
 
-const Convert: React.FC<ConvertProps> = ({
-  xvsTotalWei,
-  xvsToVrtRate,
-  vrtLimitUsedWei,
-  vrtLimitWei,
+const Convert: React.FC<IConvertProps> = ({
+  xvsToVrtConversionRatio,
   vrtConversionEndTime,
+  userVrtBalanceWei,
+  convertVrtLoading,
+  userVrtEnabled,
+  convertVrt,
   walletConnected,
 }) => {
   const styles = useStyles();
   const { t, Trans } = useTranslation();
+  const { openSuccessfulTransactionModal } = useSuccessfulTransactionModal();
+  const vrtConverterProxyAddress = getContractAddress('vrtConverterProxy');
+
   const readableXvsAvailable = useConvertToReadableCoinString({
-    valueWei: xvsTotalWei,
+    valueWei: xvsToVrtConversionRatio && userVrtBalanceWei?.times(xvsToVrtConversionRatio),
     tokenId: XVS_ID,
   });
+
+  const readableUserVrtBalance = useMemo(() => {
+    const userVrtBalanceCoins =
+      userVrtBalanceWei &&
+      convertWeiToCoins({
+        valueWei: userVrtBalanceWei,
+        tokenId: VRT_ID,
+      });
+
+    return formatCoinsToReadableValue({
+      value: userVrtBalanceCoins,
+      tokenId: VRT_ID,
+      shorthand: true,
+    });
+  }, [userVrtBalanceWei?.toString()]);
```

Comment:
So after digging yesterday, I found out we should avoid using `toString` because it returns large values using exponentials (1e22). This means that in this case some updates might not happen because two large different numbers with the same amount of exponentials and the same first digit will be returned the same.
We should use `toFixed` to avoid this issue.